### PR TITLE
Oracle Exadata docs

### PR DESCRIPTION
**Pull request by Tener**
This guide has fewer options than usual, in particular with respect to the database CA, as it documents the configuration that is known to be working. I've dropped a number of includes because of that, instead inlining their contents and tailoring it to the particulars of this case.

### CODE REVIEW

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+$ lsnrctl stop
+$ lsnrctl start
+```
+
+Additionally, your Oracle Database user accounts must be configured to require a valid client certificate. If you're creating a new user:
+
+```sql
+CREATE USER alice IDENTIFIED EXTERNALLY AS 'CN=alice';
+GRANT CREATE SESSION TO alice;
+```
```
Comment (greedy52) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:150]: nit: not sure we want to mention container/service here since we are creating a local user (per container) instead of a common user.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+
+## Step 3/6. Configure Teleport.
```
Comment (greedy52) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:152]: same comment as above, it would be nice if we can avoid switching and group similar commands run by same user together.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+Service:
+```code
+$ sudo teleport db configure create \
+   -o file \
+   --token=/tmp/token \
+   --proxy=<Var name="example.teleport.sh:443" /> \
+   --name="<Var name="oracle" />" \
+   --protocol=oracle \
+   --uri="<Var name="10.20.30.40:2484" />" \
+   --ca-cert-file="var/lib/teleport/oracle-server-certificate.crt" \
```
Comment (greedy52) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:213]:
```suggestion
   --ca-cert-file="/var/lib/teleport/oracle-server-certificate.crt" \
```

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+
+</TabItem>
+</Tabs>
+
+
+## Step 5/6. (Optional) Configure Teleport to pull audit logs from Oracle Audit Trail
+
+Teleport can pull audit logs from Oracle Audit Trail.
+In order to enable this feature, you will need to configure Oracle Audit Trail and create a dedicated Teleport
+user that will be used to fetch audit events from Oracle Audit Trail.
```
Comment (greedy52) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:301]: create new include and share between the oracle guides?

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+- Preconfigured TCPS listener.
+- An existing database certificate in the grid wallet.
+- A single Oracle Exadata VM is accessible to the `opc` user through ssh with a hostname `demodb-vm`. Repeat the steps to configure additional VMs.
+- The SCAN DNS name is `demodb-vm-scan.exadatadomain.oke.oraclevcn.com`
+
+Adjust the commands to match your configuration.
+
+</Admonition>
+
+Teleport uses mutual TLS authentication with self-hosted databases. These databases must be configured with Teleport's certificate authority to be able to verify client certificates.
```
Comment (r0mant) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:77]:
```suggestion
Teleport uses mutual TLS authentication with Exadata. It must be configured with Teleport's certificate authority to be able to verify client certificates.
```

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+...
+```
+
+For each database, update the configuration in the database home. Assuming a single database called `spark`:
+
+```code
+# Switch to oracle user
+$ sudo su - oracle
+# Load env file for database 'spark'
+$ . spark.env
```
Comment (r0mant) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:128]: I understand "spark" is just an example database you're using but it's not entirely clear whether this `.env` file is supposed to always exist and where?

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+- Oracle Exadata server instance 19c or later.
+- The `sqlcl` [Oracle client](https://www.oracle.com/pl/database/sqldeveloper/technologies/sqlcl/) installed and added to your system's `PATH` environment variable or any GUI client that supports JDBC
+  Oracle thin client.
+
+
+## Step 1/6. Create a Teleport token and user
```
Comment (r0mant) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:30]: Can this step be moved after "Configure Oracle Exadata"? Otherwise the flow of the guide is switching between Teleport and database configuration. I don't think we're creating anything in this step that will be needed for the database setup.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 8]
```diff
@@ -0,0 +1,428 @@
...
+Additionally, your Oracle Database user accounts must be configured to require a valid client certificate. If you're creating a new user:
+
+```sql
+CREATE USER alice IDENTIFIED EXTERNALLY AS 'CN=alice';
+GRANT CREATE SESSION TO alice;
+```
+
+## Step 3/6. Configure Teleport.
+
+Export the database certificate from Oracle. Teleport will expect this certificate when connecting to the database.
```
Comment (r0mant) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:154]: Is this CA certificate or something else?

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 9]
```diff
@@ -0,0 +1,428 @@
...
+```code
+# Connect to the VM
+$ ssh opc@demodb-vm
+# Switch to grid user
+$ sudo su - grid
+# Export the database certificate. Update the "-dn" parameter to match your actual setup.
+$ orapki wallet export -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets" -dn "CN=demodb-vm-scan.exadatadomain.oke.oraclevcn.com"  -cert /tmp/oracle-server-certificate.crt
+```
+
+Save the `/tmp/oracle-server-certificate.crt` file to a suitable location. It will be used by the Database service when connectiong to the Oracle database.
```
Comment (r0mant) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:165]: Same question as above essentially, I may be missing some implementation details of how Oracle auth works but this makes it sound like we're going to use this certificate for connecting to the database rather than as a CA to verify cert presented by the database.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 10]
```diff
@@ -0,0 +1,428 @@
...
+```yaml
+db_service:
+  enabled: true
+  databases:
+  - name: <Var name="oracle" />
+    uri: "<Var name="10.20.30.40:2484" />"
+    protocol: oracle
+    tls:
+      mode: verify-ca
+      ca_cert_file: /var/lib/teleport/oracle-server-certificate.crt
```
Comment (r0mant) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:228]: So this is CA certificate - then per my comments above I would rephrase in a couple of places to make this more explicit.

**Review by r0mant — APPROVED** [review]
Couple of comments but approving now to not block later.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 9 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:165]: Agreed, now that you've pointed it out I can see how it can be read in a different way than the one I've intended. I'll rephrase.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 6 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:128]: It is automatically created for each database. If you have ten databases, you will find ten different `.env` files in the home directory of Oracle user.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 11]
```diff
@@ -0,0 +1,428 @@
...
+Enable TLS auth for grid listener by editing the configuration.
+
+```code
+# Edit listener.ora to read
+# SSL_CLIENT_AUTHENTICATION = TRUE
+$ vi $ORACLE_HOME/network/admin/listener.ora
+...
+```
+
+For each database, update the configuration in the database home. Assuming a single database called `spark`:
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:122]: I meant another database on this particular server, but really this modification needs to be done per database home, which may be used by multiple databases. Sourcing env for particular database sets ORACLE_HOME, however, so you can edit `$ORACLE_HOME/network/admin/sqlnet.ora` afterwards.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 12]
```diff
@@ -0,0 +1,428 @@
...
+# Edit listener.ora to read
+# SSL_CLIENT_AUTHENTICATION = TRUE
+# SQLNET.AUTHENTICATION_SERVICES = (ALL)
+$ vi $ORACLE_HOME/network/admin/sqlnet.ora
+...
+```
+
+Repeat for additional databases as needed.
+
+Once finished, restart the listener using the `grid` user.
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:138]: I'll look into it, although I assumed switching is not very hard, being just a single `sudo su - <username>` away.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 4 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:301]: Not sure if its worth it at this moment? We only have two guides, and this section doesn't match 1-1 with the other one as the variables are different here, as we are using IP addresses in this one.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 13]
```diff
@@ -0,0 +1,428 @@
...
+$ orapki wallet add -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets" -trusted_cert -cert /tmp/teleport-db-client-ca.crt
+```
+
+Enable TLS auth for grid listener by editing the configuration.
+
+```code
+# Edit listener.ora to read
+# SSL_CLIENT_AUTHENTICATION = TRUE
+$ vi $ORACLE_HOME/network/admin/listener.ora
+...
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:119]: Fair enough, I used `vi` as we know it is generally available, but another editor may be in place instead.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 14]
```diff
@@ -0,0 +1,428 @@
...
+
+</Admonition>
+
+Teleport uses mutual TLS authentication with self-hosted databases. These databases must be configured with Teleport's certificate authority to be able to verify client certificates.
+
+Login to Teleport on your machine.
+
+```code
+$ tsh login --proxy=<Var name="example.teleport.sh" /> --user=alice
+```
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:83]: Sounds good. I think we should update the other Oracle guide in a similar way.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 7 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:30]: I didn't want to deviate from the pattern set by other guides, but I agree with you. Reorganising the steps will lead to less context switching in the guide.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 15]
```diff
@@ -0,0 +1,428 @@
...
+- Oracle Exadata server instance 19c or later.
+- The `sqlcl` [Oracle client](https://www.oracle.com/pl/database/sqldeveloper/technologies/sqlcl/) installed and added to your system's `PATH` environment variable or any GUI client that supports JDBC
+  Oracle thin client.
+
+
+## Step 1/6. Create a Teleport token and user
+
+(!docs/pages/includes/tctl-token.mdx serviceName="Database" tokenType="db" tokenFile="/tmp/token"!)
+
+<Admonition type="tip">
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: The doc now includes `docs/pages/includes/database-access/create-user.mdx`.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 16]
```diff
@@ -0,0 +1,428 @@
...
+- A single Oracle Exadata VM is accessible to the `opc` user through ssh with a hostname `demodb-vm`. Repeat the steps to configure additional VMs.
+- The SCAN DNS name is `demodb-vm-scan.exadatadomain.oke.oraclevcn.com`
+
+Adjust the commands to match your configuration.
+
+</Admonition>
+
+Teleport uses mutual TLS authentication with self-hosted databases. These databases must be configured with Teleport's certificate authority to be able to verify client certificates.
+
+Login to Teleport on your machine.
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Done.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 14 above)
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: `tctl` is now in prerequisites.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 13 above)
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Done.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Rephrased this section for clarity.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 12 above)
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Done. I still wish we could add some sort of context to the code blocks to incidate the user and machine (e.g. `grid@exadata-vm` or `oracle@exadata-vm` or `alice@workstation`), but I don't think our current doc engine allows for this.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 2 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Reordered.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 4 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: I'm leaving it as it is for now.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 17]
```diff
@@ -0,0 +1,428 @@
...
+
+| Flag         | Description                                                                                                                              |
+|--------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| `--roles`    | List of roles to assign to the user. The builtin `access` role allows them to connect to any database server registered with Teleport.   |
+| `--db-users` | List of database usernames the user will be allowed to use when connecting to the databases. A wildcard allows any user.                 |
+| `--db-names` | List of logical databases (aka schemas) the user will be allowed to connect to within a database server. A wildcard allows any database. |
+
+<Admonition type="warning">
+  Database names are only enforced for PostgreSQL and MongoDB databases.
+</Admonition>
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: I agree this is not applicable to Oracle, but database guides tend to just include `docs/pages/includes/database-access/create-user.mdx`, which has all these details.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 6 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: I've rephrased this section to explain what is an example and what are expectations.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 7 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Reordered.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 8 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Rephrased for clarity.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 18]
```diff
@@ -0,0 +1,411 @@
...
+# SQL> select * from sys.employees;
+# 
+# NAME           AGE 
+# ___________ ______ 
+# John Doe        30 
+# Cat              5 
+# 
+# SQL> exit
+# Disconnected from Oracle Database 19c EE Extreme Perf Release 19.0.0.0.0 - # Production
+# Version 19.24.0.0.0
```
Comment (greedy52) [inline docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:391]: nit: 
```suggestion
# SQL>
```

**Review by greedy52 — COMMENTED** [review]
Overall LGTM. Left a few nits. And still have the question on whether this should be in `cloud-hosted` vs `self-hosted`

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 19]
```diff
@@ -0,0 +1,411 @@
...
+$ lsnrctl start
+```
+
+## Step 2/6. Collect database configuration data.
+
+Export the built-in database certificate from Oracle. Teleport will use this certificate to verify the database connection.
+
+```code
+# Export the database certificate. Update the "-dn" parameter to match your actual setup.
+$ orapki wallet export -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets" -dn "CN=<Var name="demodb-vm-scan.exadatadomain.oke.oraclevcn.com" />" -cert /tmp/oracle-server-certificate.crt
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:134]: You can probably just peek at the wallet with:

```
$ orapki wallet display -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets"
```

But it would be good to test on a live server first, which we don't have right now.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 20]
```diff
@@ -0,0 +1,411 @@
...
+```code
+$ sudo teleport db configure create \
+   -o file \
+   --token=/tmp/token \
+   --proxy=<Var name="example.teleport.sh:443" /> \
+   --name="<Var name="oracle" />" \
+   --protocol=oracle \
+   --uri="<Var name="10.20.30.40:2484" />" \
+   --ca-cert-file="/var/lib/teleport/oracle-server-certificate.crt" \
+   --labels=env=dev
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:192]: You are right, we don't even support it here. There is no relevant flag available nor is there a place in the template:
```
    tls:
      {{- if .DatabaseCACertFile }}
      ca_cert_file: "{{ .DatabaseCACertFile }}"
      {{- end }}
      {{- if .DatabaseTrustSystemCertPool }}
      trust_system_cert_pool: {{ .DatabaseTrustSystemCertPool }}
      {{- end }}
```

We should probably add this flag, but for the time being I'll add a note to set it manually.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 21]
```diff
@@ -0,0 +1,428 @@
+---
+title: Database Access with Oracle Exadata
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:2]: I'll move it to cloud hosted section.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 22]
```diff
@@ -0,0 +1,411 @@
...
+<TabItem label="Static config">
+```yaml
+db_service:
+  enabled: "yes"
+  databases:
+  - name: <Var name="oracle" />
+    protocol: "oracle"
+    uri: "<Var name="10.20.30.40:2484" />"
+    oracle:
+      audit_user: "teleport"
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx:318]: Given that `tls.ca_cert_file: /var/lib/teleport/oracle-server-certificate.crt` is a reference to a local file, using a dynamic resource will be somewhat challenging here. I've dropped the `dynamic resource` tab to simplify the overall guide.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx [hunk 23]
```diff
@@ -0,0 +1,411 @@
...
+```
+
+Repeat these steps for each additional database or database home as required.
+
+### Configure Oracle user account
+
+Your Oracle user accounts must be configured to require a valid client certificate. 
+
+```sql
+ALTER USER alice IDENTIFIED EXTERNALLY AS 'CN=alice';
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: I've added tabs (new/existing user), and added a relevant warning for existing user.

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 20 above)
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Raised a feature request: https://github.com/gravitational/teleport/issues/45949

File: docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx (hunk 22 above)
[the remark this answers is not in the document]
Comment (greedy52) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: > I've dropped the dynamic resource tab to simplify the overall guide.
sounds good. 

> Given that tls.ca_cert_file: /var/lib/teleport/oracle-server-certificate.crt is a reference to a local file, using a dynamic resource will be somewhat challenging here.

This is not true though, you put the content of the pem for `ca_cert` (which is defined by the db spec). file reference is a static config thing

[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx]: Ah, you are right, somehow I have forgotten about this detail.

File: docs/pages/enroll-resources/database-access/enroll-managed-databases/oracle-exadata.mdx [hunk 24]
```diff
@@ -0,0 +1,399 @@
+---
+title: Database Access with Oracle Exadata
+description: How to configure Teleport database access with Oracle Exadata.
+---
+
+(!docs/pages/includes/database-access/self-hosted-introduction.mdx dbType="Oracle Exadata"!)
```
Comment (smallinsky) [inline docs/pages/enroll-resources/database-access/enroll-managed-databases/oracle-exadata.mdx:6]: This will be quite confusing. The guide is added to "Enroll Cloud-Hosted Database Platforms" but 
```
self-hosted-introduction.mdx
```

Will reference to self-hosted 

 _The Teleport Database Service proxies traffic from database clients to **self-hosted** databases in your infrastructure. Teleport maintains a certificate authority for database clients. You configure your database to trust the Teleport database client CA, and the Teleport Database Service presents certificates signed by this CA when proxying user traffic. With this setup, there is no need to store long-lived credentials for **self-hosted** databases.

Meanwhile, the Teleport Database Service verifies **self-hosted** databases by checking their TLS certificates against either the Teleport database CA or a custom CA chosen by the user_

File: docs/pages/enroll-resources/database-access/enroll-managed-databases/oracle-exadata.mdx [hunk 25]
```diff
@@ -0,0 +1,399 @@
...
+```
+
+For example, if there is a database named `spark`, the `oracle` user should have access to the automatically generated `/home/oracle/spark.env` file, which contains database-specific environment variables. After sourcing this file, the path `$ORACLE_HOME/network/admin/sqlnet.ora` will point to the `sqlnet.ora` file for the `spark` database and its associated Oracle home.
+
+```code
+# Load environment variables for the 'spark' database
+$ . spark.env 
+
+# Edit the configuration file at $ORACLE_HOME/network/admin/sqlnet.ora
+...
```
Comment (smallinsky) [inline docs/pages/enroll-resources/database-access/enroll-managed-databases/oracle-exadata.mdx:67]: Just to double check ignore if that is not correct. 

When I have playing with Oracle Gird I have been using: 

`. oraenv` 

that allow to select arbitrary SID 

Does the . spark.env configuration way will work with multiple spark_PDB* ?

**Review by smallinsky — APPROVED** [review]

**Review by greedy52 — APPROVED** [review]

File: docs/pages/enroll-resources/database-access/enroll-managed-databases/oracle-exadata.mdx (hunk 24 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-managed-databases/oracle-exadata.mdx:6]: Good catch, I'll make an update.

File: docs/pages/enroll-resources/database-access/enroll-managed-databases/oracle-exadata.mdx (hunk 25 above)
Comment (Tener) [reply docs/pages/enroll-resources/database-access/enroll-managed-databases/oracle-exadata.mdx:67]: There was no `oraenv` file for Oracle Exadata, just a single `<foo>.env` file for each attached database. Pretty sure you can just override the variables yourself to whatever values you prefer, however. You don't _need_ the env files, you could provide these values yourself if you wanted to.
